frontend/src/components/ViewActivities/activityfns.js

```diff
@@ -0,0 +1,75 @@
+import * as DBFIELDS from '../../constants/database.js';
+import app from '../Firebase';
+
+const db = app.firestore();
+
+/**
+ * Gets the list of activities from the server. 
+ * @param {string} tripId The trip ID.
+ */
+export async function getActivityList(tripId) {
+  return new Promise(function(resolve, reject) {
+    let tripActivities = [];
+    
+    db.collection(DBFIELDS.COLLECTION_TRIPS).doc(tripId)
+        .collection(DBFIELDS.COLLECTION_ACTIVITIES).get()
+        .then(querySnapshot => {
+          querySnapshot.forEach(doc => {
+        let data = doc.data();
+        data['id'] = doc.id;
+        
+        // TODO: if start date != end date, split into 2 days. (#37)
+
+        // Eliminate nanoseconds, convert to milliseconds.
+        data[DBFIELDS.ACTIVITIES_START_TIME] =
+          data[DBFIELDS.ACTIVITIES_START_TIME]['seconds'] * 1000;         
+        data[DBFIELDS.ACTIVITIES_END_TIME] = 
+          data[DBFIELDS.ACTIVITIES_END_TIME]['seconds'] * 1000;
+
+        tripActivities.push(data);
+      })
+    }).catch(error => {
+      tripActivities = null;
+    }).then( () => resolve(tripActivities) );
+  })
+}
+
+/**
+ * Sort a list of trip activities by date. 
+ * @param {Array} tripActivities Array of activities.
+ * @returns List of trip activities in the form
+ * [ ['MM/DD/YYYY', [activities on that day]], ...] in chronological order by date.
+ */
+export function sortByDate(tripActivities) {
+  let activities = new Map(); // { MM/DD/YYYY: [activities] }.
+  for (let activity of tripActivities) {
+    const activityDate = new Date(activity[DBFIELDS.ACTIVITIES_START_TIME]);
+    const dateKey = activityDate.toLocaleDateString()
+    if (activities.has(dateKey)) {
+      activities.get(dateKey).add(activity);
+    } else {
+      activities.set(dateKey, new Set([activity]));
+    }
+  }
+
+  // Sort activities by date.
+  let activitiesSorted = Array.from(activities).sort(compareActivities);
+  
+  return activitiesSorted;
```

Comment:
Nit: can do this if you want to
```suggestion
  return Array.from(activities).sort(compareActivities);
```